types/chromedriver/index.d.ts

```diff
@@ -9,6 +10,10 @@ import { ChildProcess } from 'child_process';
 
 export const path: string;
 export const version: string;
-export function start(args?: ReadonlyArray<string>): ChildProcess;
+
+export function start(args: ReadonlyArray<string>, returnPromise: true): Promise<ChildProcess>;
```

Comment:
I think the `args` parameter still needs to be optional. I know you can't put `?` on it, but perhaps make it something like `ReadonlyArray<string> | undefined | null`. Maybe TypeScript has a better way to do this that's not so verbose.

Reply:
unfortunately, that does not solve the issue. Imagine calling the so-called function with:

`const promiseWithNoArgs = chromeDriver.start(true);`

this would lead to 

`S2345: Argument of type 'true' is not assignable to parameter of type 'readonly string[] | undefined'.`

IMHO the issue is with the design of the interface. I would have defined two interfaces:

`startPromisified` and `start`

Reply:
With the typing I suggested, the caller would still be required to pass _something_ for the first parameter: `chomeDriver.start(null, true)` is an example.

Reply:
well, what's the benefit of calling the method with null rather than `chromeDriver.start([], true)`?

Reply:
I don't know what the benefit is, but the module was designed to handle it: https://github.com/giggio/node-chromedriver/blob/master/lib/chromedriver.js

Reply:
done!